javascript/javascript1/week2/main.js

```diff
@@ -0,0 +1,28 @@
+//Step 3: Smart-ease - Goes Global!
+//Flight booking fullname function
+
+function getFullname(
+  firstname = "FirstName",
+  surname = "LastName",
+  useformalName,
+  woman
+) {
+  if (useformalName === true) {
+    if (woman === true) {
+      console.log('"Lady ' + firstname + " " + surname + '"');
+    } else {
+      console.log('"Lord ' + firstname + " " + surname + '"');
+    }
+  } else {
+    if (firstname === " " || surname === " ") {
+      firstname = "FirstName";
+      surname = "LastName";
+      console.log('"' + firstname + " " + surname + '"');
+    } else {
+      console.log('"' + firstname + " " + surname + '"');
+    }
+  }
+}
+
+const fullname1 = getFullname("Aaditri", "Sarkar", true, true);
+const fullname2 = getFullname(" ");
```

Comment:
Good work 👌

**A few points to consider** 
- The naming convention. If you want to play around, you could consider naming the variable 'woman', 'gender' instead. Then you would also have to change the type of the variable and the structure of your if else statements. 
- It's worth considering, when you want to return something, or log it. In your solution you are not returning values, but rather logging them. 
You could for instance write;
else {
return '"' + firstname + " " + surname + '"'}

And then wrap your fullname1 & 2 variables in console.log() to check the values. Just some things to consider. Overall really great work 🙌

Reply:
Thanks a lot. I have modified it as per your suggestion.